File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/MinimumValidationWithSignedIntegerTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.exceptions.InvalidTypeException;
import org.openapijsonschematools.client.schemas.MapMaker;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class MinimumValidationWithSignedIntegerTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void testBoundaryPointWithFloatIsValidPasses() {
        // boundary point with float is valid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        schema.validate(
            -2.0d,
            configuration
        );
    }

    @Test
    public void testBoundaryPointIsValidPasses() {
        // boundary point is valid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        schema.validate(
            -2L,
            configuration
        );
    }

    @Test
    public void testIntBelowTheMinimumIsInvalidFails() {
        // int below the minimum is invalid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        try {
            schema.validate(
                -3L,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }

    @Test
    public void testPositiveAboveTheMinimumIsValidPasses() {
        // positive above the minimum is valid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        schema.validate(
            0L,
            configuration
        );
    }

    @Test
    public void testNegativeAboveTheMinimumIsValidPasses() {
        // negative above the minimum is valid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        schema.validate(
            -1L,
            configuration
        );
    }

    @Test
    public void testIgnoresNonNumbersPasses() {
        // ignores non-numbers
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        schema.validate(
            "x",
            configuration
        );
    }

    @Test
    public void testFloatBelowTheMinimumIsInvalidFails() {
        // float below the minimum is invalid
        final var schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance();
        try {
            schema.validate(
                -2.0001d,
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException | InvalidTypeException ignored) {
            ;
        }
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/MinimumValidationWithSignedIntegerTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class MinimumValidationWithSignedIntegerTest {
    companion object {
        val configuration = SchemaConfiguration(JsonSchemaKeywordFlags.Builder().format().build())
    }

    @Test
    fun testBoundaryPointWithFloatIsValidPasses() {
        // boundary point with float is valid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        schema.validate(
            -2.0,
            configuration
        )
    }

    @Test
    fun testBoundaryPointIsValidPasses() {
        // boundary point is valid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        schema.validate(
            -2,
            configuration
        )
    }

    @Test
    fun testIntBelowTheMinimumIsInvalidFails() {
        // int below the minimum is invalid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    -3,
                    configuration
                )
            }
        )
    }

    @Test
    fun testPositiveAboveTheMinimumIsValidPasses() {
        // positive above the minimum is valid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        schema.validate(
            0,
            configuration
        )
    }

    @Test
    fun testNegativeAboveTheMinimumIsValidPasses() {
        // negative above the minimum is valid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        schema.validate(
            -1,
            configuration
        )
    }

    @Test
    fun testIgnoresNonNumbersPasses() {
        // ignores non-numbers
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        schema.validate(
            "x",
            configuration
        )
    }

    @Test
    fun testFloatBelowTheMinimumIsInvalidFails() {
        // float below the minimum is invalid
        val schema = MinimumValidationWithSignedInteger.MinimumValidationWithSignedInteger1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    -2.0001,
                    configuration
                )
            }
        )
    }
}
